api/src/services/submission-service.ts

```diff
@@ -381,4 +390,23 @@ export class SubmissionService extends DBService {
   ): Promise<{ submission_observation_id: number }> {
     return this.submissionRepository.insertSubmissionObservationRecord(submissonObservation);
   }
+
+  /**
+   * Retrieves an array of datasets related to the given dataset.
+   *
+   * @param {string} datasetId
+   * @return {*}  {Promise<RelatedDataset[]>}
+   * @memberof SubmissionService
+   */
+  async findRelatedDatasetsByDatasetId(datasetId: string): Promise<RelatedDataset[]> {
+    const emlJson = await this.getSubmissionRecordEMLJSONByDatasetId(datasetId);
+
+    return emlJson?.['eml:eml']?.dataset?.project?.relatedProject?.map((relatedProject: any) => {
```

Comment:
I would suggest using a json path query to find the relatedProject nodes, and then parse the attributes from them in JS like normal.

The main reason is that related projects can technically also have related projects (and so on), and so we can't assume it will only be 1 level deep.

Reply:
Was definitely considering this! Ideally there would be a query to just return any `<relatedProejct>` nodes 🤔